packages/twenty-front/src/modules/object-record/record-merge/components/MergeRecordsContainer.tsx

```diff
@@ -0,0 +1,207 @@
+import { useRecoilComponentValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValueV2';
+
+import { useMergeManyRecords } from '@/object-record/hooks/useMergeManyRecords';
+import { mergeSettingsState } from '@/object-record/record-merge/states/mergeSettingsState';
+import { MergeRecordsTabId } from '@/object-record/record-merge/types/MergeRecordsTabId';
+import { useUpsertRecordsInStore } from '@/object-record/record-store/hooks/useUpsertRecordsInStore';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { useSnackBar } from '@/ui/feedback/snack-bar-manager/hooks/useSnackBar';
+import { useEffect, useState } from 'react';
+import { useNavigateApp } from '~/hooks/useNavigateApp';
+import { useMergeRecordsContainerTabs } from '../hooks/useMergeRecordsContainerTabs';
+import { MergePreviewTab } from './MergePreviewTab';
+import { MergeRecordsFooter } from './MergeRecordsFooter';
+import { MergeRecordTab } from './MergeRecordTab';
+import { MergeSettingsTab } from './MergeSettingsTab';
+
+import { AppPath } from '@/types/AppPath';
+import { useLingui } from '@lingui/react/macro';
+
+const StyledShowPageRightContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: start;
+  width: 100%;
+  height: 100%;
+  overflow: auto;
+`;
+
+const StyledTabList = styled(TabList)`
+  background-color: ${({ theme }) => theme.background.secondary};
+  padding-left: ${({ theme }) => theme.spacing(2)};
+`;
+
+const StyledContentContainer = styled.div<{ isInRightDrawer: boolean }>`
+  flex: 1;
+  overflow-y: auto;
+  background: ${({ theme }) => theme.background.primary};
+  padding-bottom: ${({ theme, isInRightDrawer }) =>
+    isInRightDrawer ? theme.spacing(16) : 0};
+`;
+
+type MergeRecordsContainerProps = {
+  componentInstanceId: string;
+  objectNameSingular: string;
+  selectedRecords: ObjectRecord[];
+  loading?: boolean;
+  isInRightDrawer?: boolean;
+};
+
+export const MergeRecordsContainer = ({
+  componentInstanceId,
+  objectNameSingular,
+  selectedRecords,
+  loading = false,
+  isInRightDrawer = false,
+}: MergeRecordsContainerProps) => {
+  const mergeSettings = useRecoilValue(mergeSettingsState);
+  const setMergeSettings = useSetRecoilState(mergeSettingsState);
+
+  const activeTabId = useRecoilComponentValueV2(
+    activeTabIdComponentState,
+    componentInstanceId,
+  );
+
+  const [mergePreviewRecord, setMergePreviewRecord] =
+    useState<ObjectRecord | null>(null);
+  const [isGeneratingPreview, setIsGeneratingPreview] = useState(false);
+
+  const { mergeManyRecords, loading: isMerging } = useMergeManyRecords({
+    objectNameSingular,
+  });
+  const { upsertRecords } = useUpsertRecordsInStore();
+
+  const navigate = useNavigateApp();
+
+  const { t } = useLingui();
+  const { tabs } = useMergeRecordsContainerTabs(selectedRecords, loading);
+  const { enqueueSuccessSnackBar, enqueueErrorSnackBar } = useSnackBar();
+
+  useEffect(() => {
```

Comment:
In the codebase we tend to isolate effects into dedicated separate components, xxxEffect

I think working on the second feature which is adding a button on the record show page might force you in the direction of abstracting things a big more